lib/posthog/event.ex

```diff
@@ -149,12 +149,19 @@ defmodule Posthog.Event do
   end
 
   @doc false
-  defp deep_stringify_keys(term) when is_map(term) do
+  defp deep_stringify_keys(term) when is_non_struct_map(term) do
     term
     |> Enum.map(fn {k, v} -> {to_string(k), deep_stringify_keys(v)} end)
     |> Enum.into(%{})
   end
 
+  defp deep_stringify_keys(term) when is_struct(term) do
+    case String.Chars.impl_for(term) do
+      nil -> Map.from_struct(term)
+      _ -> to_string(term)
+    end
+  end
```

Comment:
How do you feel about always calling Map.from_struct in that case? Would that break your use-case?

```suggestion
  defp deep_stringify_keys(term) when is_struct(term) do
    term |> Map.from_struct |> deep_stringify_keys
  end

  defp deep_stringify_keys(term) when is_map(term) do
    term
    |> Enum.map(fn {k, v} -> {to_string(k), deep_stringify_keys(v)} end)
    |> Enum.into(%{})
  end
```

Reply:
Yep, that would work too. I think that actually would properly match our behavior before the upgrade.